app/controllers/templates_controller.rb

```diff
@@ -35,4 +36,19 @@ def resource_params
   def _prefixes
     @_prefixes ||= super + ['valhalla/base']
   end
+
+  def load_fields
+    fields.each do |field|
+      case field.attribute_name
+      when 'subject'
+        @subject = field.vocabulary.categories
+      else
+        instance_variable_set("@#{field.attribute_name}", field.vocabulary.terms)
+      end
+    end
+  end
+
+  def fields
+    @fields ||= find_parent.decorate.fields
```

Comment:
Not blocking, but are there any general guidelines for when related resources should be retrieved from an instance variable and dynamically decorated, or when the decorate resource should, itself, be stored in the instance var.?